### ISSUE TITLE
Twitter Adapter : Invalid activity-missing channelId

### ISSUE THREAD

**Issue by marcasmar94** [BotBuilderCommunity/botbuilder-community-dotnet#140]
- Using Microsoft Bot Framework v4 ( Virtual Assistant Template )

        - Microsoft.Bot.Builder 4.5.1
        - Microsoft.Bot.Connector 4.5.1

I am receiving a 
### **System.ArgumentNullException**:

 **'Value cannot be null. Parameter name: invalid activity-missing channelId'**

----------------------------**StackTrace**----------------------------

 at **Microsoft.Bot.Builder.ConversationState.GetStorageKey**(ITurnContext turnContext) in d:\a\1\s\libraries\Microsoft.Bot.Builder\ConversationState.cs:line 29
   at Microsoft.Bot.Builder.BotState.<LoadAsync>d__4.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\BotState.cs:line 65
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Bot.Builder.BotState.BotStatePropertyAccessor`1.<GetAsync>d__7.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\BotState.cs:line 323
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Bot.Builder.Dialogs.DialogSet.<CreateContextAsync>d__9.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogSet.cs:line 92
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at MockBotAlfa.Bots.DialogBot`1.<OnTurnAsync>d__3.MoveNext() in C:\Users\Mark\source\repos\MockBotAlfaSolution\MockBotAlfa\Bots\DialogBot.cs:line 39
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Bot.Builder.Community.Twitter.Adapter.Hosting.WebhookMiddleware.<OnDirectMessageReceived>d__7.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
----------------------------**End StackTrace**----------------------------

The problem is when trying to get the **ChannelID** 

###  **ConversationState.cs** from Microsoft.Bot.Builder 

```
 protected override string GetStorageKey(ITurnContext turnContext)
        {
            var channelId = turnContext.Activity.ChannelId ?? throw new ArgumentNullException("invalid activity-missing channelId");
            var conversationId = turnContext.Activity.Conversation?.Id ?? throw new ArgumentNullException("invalid activity-missing Conversation.Id");
           return $"{channelId}/conversations/{conversationId}";
        }
 
```

**Comment by tmarkovski** [comment]
Looks like this can be solved by adding `ChannelId = "twitter"` to https://github.com/BotBuilderCommunity/botbuilder-community-dotnet/blob/c65664f914f1e5c47d78de9d00c6a52e2a1e43c1/libraries/Bot.Builder.Community.Adapters.Twitter/Hosting/WebhookMiddleware.cs#L63

**Comment by tmarkovski** [comment]
This issue is resolved with #142 and can be closed.

**Comment by arafattehsin** [comment]
Thanks! @tmarkovski - closing this issue.
